### PR TITLE
POR-1856 show error on app update failure

### PR DESCRIPTION
**Pull request by ianedwards**
## What does this PR do?

- Shows button error when validate/apply fails on app update

### CODE REVIEW

File: dashboard/src/main/home/app-dashboard/app-view/AppDataContainer.tsx [hunk 1]
```diff
@@ -293,23 +304,34 @@ const AppDataContainer: React.FC<AppDataContainerProps> = ({ tabParam }) => {
         appName: latestProto.name,
         errorStackTrace: stack,
       });
+
+      if (err.response?.data?.message) {
+        setError("app", {
+          message: `App updated failed: ${err.response.data.message}`,
```
Comment (d-g-town) [inline dashboard/src/main/home/app-dashboard/app-view/AppDataContainer.tsx:310]: `updated` -> `update`

File: dashboard/src/main/home/app-dashboard/app-view/AppDataContainer.tsx [hunk 2]
```diff
@@ -333,22 +355,45 @@ const AppDataContainer: React.FC<AppDataContainerProps> = ({ tabParam }) => {
...
+  const buttonStatus = useMemo(() => {
+    if (isSubmitting) {
+      return "loading";
+    }
+
+    if (Object.keys(errors).length > 0) {
+      return errors.app?.message ? (
+        <Error message={errors.app.message} />
+      ) : (
+        <Error message="App update failed!" />
```
Comment (d-g-town) [inline dashboard/src/main/home/app-dashboard/app-view/AppDataContainer.tsx:367]: maybe no exclamation point, and direct them to contact support?

File: dashboard/src/main/home/app-dashboard/app-view/tabs/Environment.tsx [hunk 3]
```diff
@@ -74,7 +63,6 @@ const Environment: React.FC<Props> = ({ latestSource }) => {
         revision={previewRevision ? previewRevision : latestRevision} // get versions of env groups attached to preview revision if set
         baseEnvGroups={baseEnvGroups}
         latestSource={latestSource}
-        servicesFromYaml={servicesFromYaml}
```
Comment (d-g-town) [inline dashboard/src/main/home/app-dashboard/app-view/tabs/Environment.tsx:77]: why is this being removed?

File: dashboard/src/main/home/app-dashboard/app-view/tabs/Environment.tsx [hunk 4]
```diff
@@ -11,23 +11,24 @@ import api from "shared/api";
...
     latestProto,
     clusterId,
     projectId,
     previewRevision,
-    servicesFromYaml,
     attachedEnvGroups,
   } = useLatestRevision();
   const {
-    formState: { isSubmitting, errors },
+    formState: { isSubmitting },
```
Comment (d-g-town) [inline dashboard/src/main/home/app-dashboard/app-view/tabs/Environment.tsx:31]: why is errors being removed?

**Review by d-g-town — APPROVED** [review]

File: dashboard/src/main/home/app-dashboard/app-view/tabs/Environment.tsx (hunk 3 above)
Comment (ianedwards) [reply dashboard/src/main/home/app-dashboard/app-view/tabs/Environment.tsx:77]: Turns out it wasn't being used

File: dashboard/src/main/home/app-dashboard/app-view/tabs/Environment.tsx (hunk 4 above)
Comment (ianedwards) [reply dashboard/src/main/home/app-dashboard/app-view/tabs/Environment.tsx:31]: I consolidated all the error handling into the top-level component, so the child components just get the derived `buttonStatus`

**Pull request merged**